### PR TITLE
Allow `create.specialVerbs` to be customized

### PR DESCRIPTION
**Pull request by rosspeoples**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This allows allows downstream distributions to customize the behavior of `kubectl create` with non-k8s native custom resources.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_role.go [hunk 1]
```diff
@@ -112,6 +112,14 @@ var (
 	}
 )
 
+// AddSpecialVerb allows the addition of items to the `specialVerbs` map for non-k8s native resources.
+func AddSpecialVerb(verb string, gr schema.GroupResource) {
+	if resources, ok := specialVerbs[verb]; ok {
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_role.go:117]: this doesn't work correctly if `verb` is not already in the `specialVerbs` map

**Pull request merged**